src/components/app/flyouts/settings/settings.js

```diff
@@ -0,0 +1,149 @@
+
+import './settings.css';
+
+const Section = Flyout.Section;
+
+export class Settings extends LinkedComponent {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      desiredSimulationState: this.props.isSimulationEnabled,
+      logoFile: undefined,
+      applicationName: '',
+      loading: false
+    };
+
+    // this.eTag = new BehaviorSubject(undefined);
+    // this.unmount = new Subject();
+    // this.eTagStream = this.eTag.filter(_ => _);
+    const { t } = this.props;
+
+    // Helper objects for choosing the correct label for the simulation service toggle input
+    this.desiredSimulationLabel = {
+      true: t('settingsFlyout.start'),
+      false: t('settingsFlyout.stop')
+    };
+    this.currSimulationLabel = {
+      true: t('settingsFlyout.flowing'),
+      false: t('settingsFlyout.stopped')
+    };
+
+    this.applicationName = this.linkTo('applicationName')
+      .map(value => value.length === 0 ? undefined : value);
+
+    this.props.getSimulationStatus();
+  }
+
+  componentWillReceiveProps({ isSimulationEnabled }) {
+    if(this.state.desiredSimulationState === undefined && isSimulationEnabled !== undefined) {
+      this.setState({
+        desiredSimulationState : isSimulationEnabled
+      });
+    }
+  }
+
+  onChange = ({ target }) => {
+    const { name, value } = target;
+    this.setState({ [name]: value });
+  };
+
+  apply = () => {
+    const { logoFile, applicationName, desiredSimulationState } = this.state;
+    if (logoFile || applicationName) {
+      var headers = {};
+      if (applicationName) {
+        headers.name = applicationName
+      }
+      if (logoFile) {
+        headers['Content-Type'] = logoFile.type;
+      } else {
+        headers['Content-Type'] = "text/plain";
+      }
+      this.props.updateLogo(logoFile, headers);
+    }
+    if(desiredSimulationState !== this.props.isSimulationEnabled) {
+      const etag = this.props.simulationEtag;
+      this.props.toggleSimulationStatus(etag, desiredSimulationState);
+    }
+    this.props.onClose();
+  };
+
+  onUpload = (file) => {
+    this.setState({
+      logoFile: file
+    });
+  };
+
+  render() {
+    const { t, onClose, theme, changeTheme, version, releaseNotesUrl, isSimulationEnabled } = this.props;
+    const nextTheme = theme === 'dark' ? 'light' : 'dark';
+    const { desiredSimulationState, loading, logoFile, applicationName } = this.state;
+    const stillInitializing = desiredSimulationState === undefined;
+    const hasChanged = !stillInitializing && (isSimulationEnabled !== desiredSimulationState
+      || logoFile !== undefined || applicationName !== '');
+    const hasSimulationChanged = !stillInitializing && (isSimulationEnabled !== desiredSimulationState)
+    const simulationLabel = hasSimulationChanged ? this.desiredSimulationLabel[desiredSimulationState] : this.currSimulationLabel[isSimulationEnabled];
+
+    return (
+      <Flyout.Container>
```

Comment:
The form elements need to be wrapped in a `<form>` tag and the apply handled `onSubmit`